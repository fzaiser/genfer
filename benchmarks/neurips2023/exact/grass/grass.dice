let cloudy  = flip 0.5 in
let rain  = if cloudy then flip 0.8 else flip 0.2 in
let sprinkler  = if cloudy then flip 0.1 else flip 0.5 in
let temp1  = flip 0.7 in
let wetRoof  = (temp1 && rain) in
let temp2  = flip 0.9 in
let temp3  = flip 0.9 in
let wetGrass  = ((temp2 && rain) || ( temp3 && sprinkler)) in
let tmp = observe wetGrass in
rain