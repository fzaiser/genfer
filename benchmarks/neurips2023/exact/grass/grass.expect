Support is a subset of: {0, ..., 1}

Computing moments...
Total measure:             Z = 0.6471000000000001
Expected value:            E = 0.7079276773296244
Standard deviation:        Ïƒ = 0.45471538460921634
Variance:                  V = 0.20676608100030758
Skewness (3rd std moment): S = -0.9145398830449426
Kurtosis (4th std moment): K = 1.83638319767985

Computing probabilities up to 2...
Unnormalized: p(0)     = 0.189
Normalized:   p(0) / Z = 0.2920723226703755
Unnormalized: p(1)     = 0.4581
Normalized:   p(1) / Z = 0.7079276773296244
Unnormalized: p(n)     <= 0.0 for all n >= 2
Normalized:   p(n) / Z <= 0.0 for all n >= 2
