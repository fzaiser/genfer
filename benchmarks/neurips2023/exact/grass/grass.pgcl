nat cloudy
nat rain
nat sprinkler
nat wetRoof
nat wetGrass

cloudy := bernoulli(0.5);
rain := 0;
sprinkler := 0;

if(cloudy = 1) {
    rain := bernoulli(0.8);
    sprinkler := bernoulli(0.1);
} else {
    rain := bernoulli(0.2);
    sprinkler := bernoulli(0.5);
}

temp1 := bernoulli(0.7);
if(temp1 = 1 & rain = 1) {
    wetRoof := 1;
} else {
    wetRoof := 0;
}
temp2 := bernoulli(0.9);
temp3 := bernoulli(0.9);
if((temp2 = 1 & rain = 1) || (temp3 = 1 & sprinkler = 1)) {
    wetGrass := 1;
} else {
    wetGrass := 0;
}

observe(wetGrass = 1);

?Pr[rain];
