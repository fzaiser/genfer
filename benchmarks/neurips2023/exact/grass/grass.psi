// flags: --dp
def main(){ // didItRain
    cloudy := flip(1/2);
    rain := 0; sprinkler := 0;

    if (cloudy){
        rain = flip(4/5);
        sprinkler = flip(1/10);
    }else{
        rain = flip(1/5);
        sprinkler = flip(1/2);
    }

    temp1 := flip(7/10);
    wetRoof := temp1 && rain;
    temp2 := flip(9/10);
    temp3 := flip(9/10);
    wetGrass := (temp2 && rain) || (temp3 && sprinkler);

    observe(wetGrass);
    return rain;
}
