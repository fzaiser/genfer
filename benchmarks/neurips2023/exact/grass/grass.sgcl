cloudy ~ Bernoulli(0.5);
rain := 0;
sprinkler := 0;

if cloudy = 1 {
    rain ~ Bernoulli(0.8);
    sprinkler ~ Bernoulli(0.1);
} else {
    rain ~ Bernoulli(0.2);
    sprinkler ~ Bernoulli(0.5);
}

temp1 ~ Bernoulli(0.7);
if temp1 = 1 && rain = 1 {
    wetRoof := 1;
} else {
    wetRoof := 0;
}
temp2 ~ Bernoulli(0.9);
temp3 ~ Bernoulli(0.9);
if (temp2 = 1 && rain = 1) || (temp3 = 1 && sprinkler = 1) {
    wetGrass := 1;
} else {
    wetGrass := 0;
}

observe(wetGrass = 1);

return rain;
