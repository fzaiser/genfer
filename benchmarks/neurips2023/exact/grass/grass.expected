0.70792767
509/719
